yt_idefix/data_structures.py

```diff
@@ -320,3 +321,116 @@ def _parse_parameter_file(self):
+    ):
+        self._definitions_header = definitions_header
+        super().__init__(
+            filename,
+            dataset_type=dataset_type,
+            unit_system=unit_system,
+            units_override=units_override,
+            geometry=geometry,
+            inifile=inifile,
+        )
+
+    def _parse_parameter_file(self):
+        self._parse_header_file()
+        super()._parse_parameter_file()
+        self._get_time()
+
+    def _parse_header_file(self):
+        """Read some metadata from header file 'definitions.h'."""
+        geom_regexp = re.compile(r"^\s*#define\s+GEOMETRY\s+([A-Z]+)")
+        unit_regexp = re.compile(r"^\s*#define\s+(UNIT_\w+)\s+(\S+)")
+        constexpr = re.compile(r"CONST_\w+")
+
+        # definitions.h is presumed to be along with data file
+        if self._definitions_header is None:
+            self._definitions_header = os.path.join(self.fullpath, "definitions.h")
+
+        if os.path.isfile(self._definitions_header):
+            with open(self._definitions_header) as fh:
+                for line in fh.readlines():
+                    geom_match = re.search(geom_regexp, line)
+                    unit_match = re.search(unit_regexp, line)
+                    if geom_match is None and unit_match is None:
+                        continue
+                    elif unit_match:
+                        unit = unit_match.group(1).lower()
+                        expr = unit_match.group(2)
+                        # The pre-defined constants in definitions should be
+                        # replace with its value
+                        expr = re.sub(constexpr, self._get_constants, expr)
+                        self.parameters[unit] = eval(expr)
+                    elif geom_match:
+                        self.geometry = geom_match.group(1).lower()
+        elif self.geometry is None:
+            raise ValueError(
+                f"Header file {self._definitions_header} couldn't be found. "
+                "The 'geometry' keyword argument must be specified."
+            )
+        else:
+            warnings.warn(
+                f"Header file {self._definitions_header} couldn't be found. "
+                "The code units are set to be 1.0 in cgs by default."
+            )
+
+    def _get_constants(self, matched):
+        """Replace matched constant string with its value"""
+        key = matched.group()
+        return str(pluto_def_constants[key])
+
+    def _get_time(self):
+        """Get current time from vtk.out."""
+        log_file = os.path.join(self.fullpath, "vtk.out")
+        match = re.search(r"\.(\d*)\.", self.parameter_filename)
+        if match is None:
+            raise RuntimeError(
+                f"Failed to parse output number from file name {self.parameter_filename}"
+            )
+        index = int(match.group(1))
+
+        self.current_time = -1
+        if os.path.isfile(log_file):
+            log_regexp = re.compile(fr"^{index}\s(\S+)")
```

Comment:
if I'm not mistaken, the second capturing group `(\S+)` is meant to represent a number (time) possibly written in scientific notation, right ?
Eventually I'd like to refactor this to utilise `inifix.ENotationIO.decode`, but I'm not requiring this for this PR (inifix is a different library that I also maintain and which is already a direct dependency to yt_idefix).

Reply:
> if I'm not mistaken, the second capturing group (\S+) is meant to represent a number (time) possibly written in scientific notation, right ?

Yes, you're right. I checked inifix and it seems like `inifix.ENotationIO.decode` cannot be called outside the package right now. So I'll leave it for future PR.

Reply:
No worries, I'll do it myself. Possibly I'll change how inifix exposes it.